### ISSUE TITLE
`master` branch gives TypeError and quits

### ISSUE THREAD

**Issue by osirisinferi** [certbot/certbot#3377]
Recently, I updated my `certbot` to the latest Github master branch (Gentoo makes this very easy). Unfortunately, `certbot` stopped working. I confirmed the error on a separate Gentoo box in VirtualBox, which is as clean as it can be, so I'm doubting some weird setting I'm having on my "production" server causes the error.

```
Root logging level set at 0
Saving debug log to /var/log/letsencrypt/letsencrypt.log
certbot version: 0.9.0.dev0
Arguments: ['-debug', '-tvv', '-i', 'apache', '-a', 'webroot', '-d', 'le-test-12.example.com', '-w', '/var/www/vhosts/le-test-12.example.com/htdocs/', '--staging']
Discovered plugins: PluginsRegistry(PluginEntryPoint#apache,PluginEntryPoint#webroot,PluginEntryPoint#null,PluginEntryPoint#manual,PluginEntryPoint#standalone)
Requested authenticator webroot and installer apache
Exiting abnormally:
Traceback (most recent call last):
  File "/usr/lib/python-exec/python2.7/certbot", line 9, in <module>
    load_entry_point('certbot==0.9.0.dev0', 'console_scripts', 'certbot')()
  File "/usr/lib64/python2.7/site-packages/certbot/main.py", line 770, in main
    return config.func(config, plugins)
  File "/usr/lib64/python2.7/site-packages/certbot/main.py", line 504, in run
    installer, authenticator = plug_sel.choose_configurator_plugins(config, plugins, "run")
  File "/usr/lib64/python2.7/site-packages/certbot/plugins/selection.py", line 187, in choose_configurator_plugins
    installer = pick_installer(config, req_inst, plugins)
  File "/usr/lib64/python2.7/site-packages/certbot/plugins/selection.py", line 31, in pick_installer
    config, default, plugins, question, (interfaces.IInstaller,))
  File "/usr/lib64/python2.7/site-packages/certbot/plugins/selection.py", line 76, in pick_plugin
    verified.prepare()
  File "/usr/lib64/python2.7/site-packages/certbot/plugins/disco.py", line 220, in prepare
    return [plugin_ep.prepare() for plugin_ep in six.itervalues(self._plugins)]
  File "/usr/lib64/python2.7/site-packages/certbot/plugins/disco.py", line 106, in prepare
    self._initialized.prepare()
  File "/usr/lib64/python2.7/site-packages/certbot_apache/configurator.py", line 191, in prepare
    self.vhosts = self.get_virtual_hosts()
  File "/usr/lib64/python2.7/site-packages/certbot_apache/configurator.py", line 573, in get_virtual_hosts
    new_vhost = self._create_vhost(path)
  File "/usr/lib64/python2.7/site-packages/certbot_apache/configurator.py", line 540, in _create_vhost
    filename = get_file_path(self.aug.get("/augeas/files%s/path" % get_file_path(path)))
  File "/usr/lib64/python2.7/site-packages/certbot_apache/configurator.py", line 1805, in get_file_path
    avail_fp = vhost_path[6:]
TypeError: 'NoneType' object has no attribute '__getitem__'

An unexpected error occurred:
TypeError: 'NoneType' object has no attribute '__getitem__'
Please see the logfiles in /var/log/letsencrypt for more details.
```

Unfortunately, I've got no real clue how to debug this.. When using `ipdb` at multiple sites in `configurator.py`, I'm not getting anywhere.

Is this a Gentoo thing or can other people confirm this error?

**Comment by jpluimers** [comment]
I finally got around playing on opensuse with https://certbot.eff.org/docs/contributing.html#running-a-local-copy-of-the-client so I could try the master branch for #3364 and #3307

Now it’s throwing `TypeError: 'NoneType' object has no attribute '__getitem__’` at me even with just `certbot` as command.
Log file is at https://gist.github.com/

**Comment by joohoi** [comment]
I got opensuse box up to try and reproduce this, but was unable to do so against the master HEAD. @jpluimers could you post your vhost configuration, so I could take a closer look into it. @osirisinferi if you have /etc/os-release file in your filesystem, could you post the contents, I have a hunch that this might have something to do with being unable to fingerprint your distro.

**Comment by jpluimers** [comment]
There's a whole bunch of files in `/etc/apache2/vhosts.d`; which ones do you need?

```
./la-perlina.nl.conf
./00-default.revue.conf
./revue.conf
./pluimers.com.conf
./continuaci.pluimers.com-ssl.conf
./continuaci.pluimers.com.conf
./continuaci.4delphi.com.conf
./continuaci.4delphi.com-ssl.conf
./www.4delphi.com.conf
./pcm.pluimers.com.conf
./pcm.pluimers.com-ssl.conf
./pcm.4delphi.com.conf
./pcm.4delphi.com-ssl.conf
./https-shell.locations.include
./vhost-ssl.template
./vhost.template
./www.pluimers.com-ssl.conf
./www.4delphi.com-ssl.conf
```

This is `/etc/os-release`:

```
NAME=openSUSE
VERSION="Tumbleweed"
VERSION_ID="20160805"
PRETTY_NAME="openSUSE Tumbleweed (20160805) (x86_64)"
ID=opensuse
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:opensuse:20160805"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
ID_LIKE="suse"
```

**Comment by osirisinferi** [comment]
@joohoi The culprit is 2e50b38.. That commit fails, while its parent a506436 works perfectly.

Fortunately! You've fixed it in (at least) 14f3710 as you already pointed out in your PR :stuck_out_tongue: 

In any event, as requested, my `/etc/os-release`:
```
NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo/Linux"
ANSI_COLOR="1;32"
HOME_URL="http://www.gentoo.org/"
SUPPORT_URL="http://www.gentoo.org/main/en/support.xml"
BUG_REPORT_URL="https://bugs.gentoo.org/"
```

**Comment by jpluimers** [comment]
Indeed: fixed.
